copylot/assemblies/photom/demo/demo_roi.py

```diff
@@ -0,0 +1,314 @@
+
+                self.curr_shape_points = []
+                self.drawing = False
+                self.curr_shape_id += 1
+            self.update()
+
+    def paintEvent(self, event: QPaintEvent) -> None:
+        """handles the paint event, drawing shapes and patterns on the widget.
+
+        Args:
+            event (QPaintEvent): the paint event that triggered the redrawing. 
+        """
+        self.draw_shapes()
+        self.draw_patterns()
+
+    def draw_shapes(self) -> None:
+        """draws all the shapes on the widget.
+        """
+        painter = QPainter(self)
+
+        for shape_id, shape in self.shapes.items():
+            if shape_id == self.selected_shape_id:
+                pen = QPen(Qt.red, 2, Qt.SolidLine)
+            else:
+                pen = QPen(Qt.black, 2, Qt.SolidLine)
+            painter.setPen(pen)
+
+            border_points = shape.border_points
+            for i in range(len(border_points) - 1):
+                painter.drawLine(border_points[i], border_points[i + 1])
+
+        if self.drawing and len(self.curr_shape_points) > 1:
+            pen = QPen(Qt.black, 2, Qt.SolidLine)
+            painter.setPen(pen)
+            for i in range(len(self.curr_shape_points) - 1):
+                painter.drawLine(self.curr_shape_points[i], self.curr_shape_points[i + 1])
+
+    def draw_patterns(self) -> None:
+        """draws all the patterns in the shapes on the widget.
+        """
+        painter = QPainter(self)
+        pen = QPen(Qt.green, 2, Qt.SolidLine)
+        painter.setPen(pen)
+
+        for shape_id, shape in self.shapes.items():
+            if shape.pattern_points:
+                pattern_points = shape.pattern_points
+                for point in pattern_points:
+                    point = QPoint(point[0], point[1])
+                    painter.drawPoint(point)
+
+class CtrlWindow(QMainWindow):
+    def __init__(self, viewer_window: QMainWindow) -> None:
+        """initializes the control window.
+
+        Args:
+            viewer_window (QMainWindow): the viewer window (main window) associated with the control window.
+        """
+        super().__init__()
+        self.viewer_window = viewer_window 
+        self.windowGeo = (500, 500, 500, 500)
+        self.buttonSize = (200, 100)
+        self.patterns = ['Bidirectional']
+        self.initUI()
+
+    def initUI(self) -> None:
+        """initializes the user interface for the control window.
+        """
+        central_widget = QWidget()
+        self.setCentralWidget(central_widget)
+        layout = QVBoxLayout(central_widget)
+
+        self.setGeometry(
+            self.windowGeo[0],
+            self.windowGeo[1],
+            self.windowGeo[2],
+            self.windowGeo[3],
+        )
+        self.setWindowTitle('Control panel')
+        self.setFixedSize(
+            self.windowGeo[2],
+            self.windowGeo[3],
+        )
+
+        # roi dropdown box
+        self.roi_dropdown= QComboBox(self)
```

Comment:
these need to be added to the control window in `gui.windows.py`

Reply:
all the new buttons and functionality should show up in the photom GUI when you enable 'drawing mode'.

Reply:
and should dissapear otherwise.